java-common/src/main/java/com/typesafe/conductr/java/ConnectionHandler.java

```diff
@@ -0,0 +1,61 @@
+package com.typesafe.conductr.java;
+
+import com.typesafe.conductr.HttpPayload;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import static com.typesafe.conductr.java.ManagedBlocker.*;
+
+/**
+ * Connection handlers provide the means to establish a connection, issue a request and then finalize
+ * the connection
+ */
+public class ConnectionHandler {
+
+    protected final static String USER_AGENT = "TypesafeConductRBundleLib";
```

Comment:
User agent should be not TypesafeConductRBundleLib, but more generic, e.g. `TypesafeConductRLib`.